Configurations/BaseCommon.xcconfig

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2022 Apple Inc. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+// 1. Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// This configuration file should be used to set defaults for all WebKit projects.
+
+#include "Sanitizers.xcconfig"
+// SDKVariant.xcconfig is included via Sanitizers.xcconfig.
+
+GCC_OPTIMIZATION_LEVEL = $(GCC_OPTIMIZATION_LEVEL_WK_OVERRIDE_$(WK_OVERRIDE_GCC_OPTIMIZATION_LEVEL));
+GCC_OPTIMIZATION_LEVEL_WK_OVERRIDE_YES = $(WK_GCC_OPTIMIZATION_LEVEL);
+// Define GCC_OPTIMIZATION_LEVEL_WK_OVERRIDE_NO in each project's Base.xcconfig.
+
+WK_GCC_OPTIMIZATION_LEVEL = $(WK_GCC_OPTIMIZATION_LEVEL_FORCED_$(WK_HAS_FORCE_OPTIMIZATION_LEVEL));
+WK_GCC_OPTIMIZATION_LEVEL_FORCED_YES = $(WK_FORCE_OPTIMIZATION_LEVEL);
```

Comment:
What's the use case for WK_FORCE_OPTIMIZATION_LEVEL? 

I'm probably misunderstanding, but I don't see anywhere in the PR where we set this. If it's intended to be set ad-hoc by users, why can't they just override GCC_OPTIMIZATION_LEVEL on the command line?

Reply:
It's a `Tools/Scripts/set-webkit-configuration` switch:

```
  --force-optimization-level=<level> Optimization level: O3, O2, O1, O0, Os, Ofast, Og, or none
```

Used by @TuomasWebKit and the JavaScriptCore team, IIRC.

Reply:
And I just realized that with this change, we're no longer specifying an `xcconfig` file on the command-line, so we can simplify this code in `Tools/Scripts/webkitdirs.pm`:

```
    if ($forceOptimizationLevel) {
        if ($asanIsEnabled || $tsanIsEnabled || $ubsanIsEnabled) {
            # Command-line Xcode variable won't override that same varible set in a command-line xcconfig file.
            push @options, "WK_FORCE_OPTIMIZATION_LEVEL=$forceOptimizationLevel";
        } else {
            push @options, "GCC_OPTIMIZATION_LEVEL=$forceOptimizationLevel";
        }
    }
```